tests/test_vmdata.py

```diff
@@ -81,3 +83,102 @@ def test_set_column_headings(df: pd.DataFrame, units: str, version: int) -> None
 
     assert vmdata.column_headers == expected_headers
     assert all([version is not vmdata.column_headers for version in vm_const.COLUMN_HEADERS.values()])
+
+
+def test_set_column_headings_invalid() -> None:
+    vmdata = VMData(
+        df=pd.DataFrame(
+            {
+                "VM OS Name": ["Windows 10", "Ubuntu 20.04", "CentOS 7"],
+                "Environments": ["Prod", "Dev", "Prod"],
+                "VM MEM": [8, 16, 32],
+                "VM Provisioned Disk": [100, 200, 300],
+            }
+        )
+    )
+
+    with pytest.raises(ValueError):
+        vmdata.set_column_headings()
+
+    assert vmdata.column_headers == {}
+
+
+@pytest.mark.usefixtures("datafile")
+@pytest.mark.parametrize("datafile", ["csv"], indirect=["datafile"])
+def test_add_extra_columns(vmdata: VMData) -> None:
+    vmdata.set_column_headings()
+    original_df = vmdata.df.copy()
+    unmodified_columns = list(set(original_df.columns).difference(set(vm_const.EXTRA_COLUMNS_DEST)))
+
+    vmdata.add_extra_columns()
+
+    assert (
+        len(
+            set(vm_const.EXTRA_WINDOWS_DESKTOP_COLUMNS + vm_const.EXTRA_WINDOWS_DESKTOP_COLUMNS).intersection(
```

Comment:
Just a question about whether or not this should be adding the same constants together?

Reply:
It should have been adding both window server and windows desktop column names together,  to match the columns dropped in https://github.com/rhtools/vminfo-parser/blob/5b964fca9cf5276cb87ebc03a180aedefd780d46/vminfo_parser/vminfo_parser.py#L215

fixed in https://github.com/rhtools/vminfo-parser/pull/9/commits/5b964fca9cf5276cb87ebc03a180aedefd780d46